pydantic/typing.py

```diff
@@ -48,11 +48,16 @@ def evaluate_forwardref(type_, globalns, localns):  # type: ignore
 
     AnyCallable = TypingCallable[..., Any]
 
-try:
-    from typing_extensions import Literal
-except ImportError:
-    Literal = None  # type: ignore
-
+if sys.version_info < (3, 8):
+    if TYPE_CHECKING:
+        from typing_extensions import Literal
+    else:  # due to different mypy warnings raised during CI for python 3.7 and 3.8
```

Comment:
Had to add this gross `if TYPE_CHECKING:` block here because the `Literal = None` line was causing an "unnecessary ignore" warning during the 3.8 build. This approach seems fine to me here, but I'm open to alternatives.

Reply:
This is a [known issue](https://github.com/python/mypy/issues/5940) in mypy with one developer commenting that `--warn-unused-ignores` is not recommended for use in CI. But barring that you can trick mypy into interpreting `Literal` as a `Union` with:

```py
if sys.version_info < (3, 8):
    try:
        from typing_extensions import Literal as _Literal
    except ImportError:
        Literal = None
    else:
        Literal = _Literal
else:
    from typing import Literal
```

Reply:
Thanks for sharing the issue.

I definitely would prefer to keep that check on in CI though; otherwise it's too easy to accidentally forget, and then end up with confusion about what is being ignored and/or ignore a future error that should have been caught.

Reply:
agreed, I hoped mypy would have a `type: ignore flexible` option, but couldn't find anything.